### PR TITLE
fix: removes addSubjectMetadata and improves Snaps Execution Environment Webview origin restriction.

### PR DESCRIPTION
**Pull request by Jonathansoufer**
## **Description**

This PR address removal of subjectMetadata in favor of SnapController triggering addSubjectMetadata itself now, and adding a more restrictive originWhiteList on Snaps Execution Environment Webview.

## **Related issues**

Fixes: Comments on the closed PR 8700 

## **Manual testing steps**

1. Go to Explore and Install any snap.

### **After**
https://github.com/MetaMask/metamask-mobile/assets/44679989/796853dd-e906-4350-abcb-3467a3c6ba26

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/core/Engine.ts
Comment (FrederikBolding) [inline app/core/Engine.ts:1]: Looks like this has some linting problems, otherwise good 👍

File: app/lib/snaps/SnapsExecutionWebView.tsx [hunk 1]
```diff
@@ -21,8 +21,6 @@ interface SnapsExecutionWebViewProps {
 let resolveGetWebView: (arg0: SnapsExecutionWebViewProps) => void;
 let rejectGetWebView: (error: NativeSyntheticEvent<WebViewError>) => void;
 
-const SNAPS_EE_URL = 'https://execution.metamask.io/webview/4.0.0/index.html';
```
Comment (FrederikBolding) [inline app/lib/snaps/SnapsExecutionWebView.tsx:24]: Why did we move this from a constant?

Comment (Jonathansoufer) [reply app/lib/snaps/SnapsExecutionWebView.tsx]: This was due my tests. Forgot to put it back. Fixed.

**Review by FrederikBolding — APPROVED** [review]

**Review by NicolasMassart — APPROVED** [review]
Looks good to me.

**Pull request merged**